cluster-api/Makefile

```diff
@@ -12,14 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: genapi genconversion
+.PHONY: genapi clean
 
-all: genapi genconversion
+APIBUILDER_GENERATORS=apiregister conversion openapi
+APIBUILDER_IMAGE_TAG=dev-cluster-api-generator
 
+all: genapi clean
+
+# Use apiserver-builder to automatically enerate API code
+# This will:
+# - Build a Docker image with the appropriate apiserver-builder version
+# - Add the current host user as a user in the container so that the code
+#   may be bind mounted with the correct permissions
+# - Execute the apiserver-builder generation process
+# - Clean up the build and tagged dev-cluster-api-generator Docker image
 genapi:
-	go install github.com/kubernetes-incubator/apiserver-builder/cmd/apiregister-gen
-	apiregister-gen -i ./pkg/apis,./pkg/apis/cluster/v1alpha1
+	docker build --build-arg DEVUSERNAME=`id -un` --build-arg DEVUSERID=`id -u` -f Dockerfile.apigen . -t ${APIBUILDER_IMAGE_TAG}
```

Comment:
Is there a way to not have to build this every time?

Reply:
Are you looking for this image to be in a registry somewhere or do you mean just locally?  The tricky part is that we need to create a user that matches the local `$USER` so that the generated code is not permissioned as root.  I figured this would be used so infrequently that it is not much of a burden to take on.